### PR TITLE
Issue #1 Create boilerplate

### LINKED ISSUE #1: Create 00-boilerplate
Empty React project with Typescript support.

You can use as starting point:

https://github.com/Lemoncode/master-frontend-lemoncode/tree/master/04-frameworks/01-react/02-base/03-webpack-react

But updating code to React 18

### PR DESCRIPTION
**Pull request by carlos-arcos**
closed #1 @carlos-arcos

### CODE REVIEW

File: 00-boilerplate/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    "babel-loader": "^8.2.3",
+    "css-loader": "^6.4.0",
+    "file-loader": "^6.2.0",
+    "html-loader": "^3.0.0",
+    "html-webpack-plugin": "^5.4.0",
+    "npm-run-all": "^4.1.5",
+    "rimraf": "^3.0.2",
+    "style-loader": "^3.3.1",
+    "typescript": "^4.4.4",
+    "url-loader": "^4.1.1",
```
Comment (brauliodiez) [inline 00-boilerplate/package.json:32]: Remove url-loader now we can use webpack asset loader

File: 00-boilerplate/package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    "@babel/cli": "^7.15.7",
+    "@babel/core": "^7.15.8",
+    "@babel/preset-env": "^7.15.8",
+    "@babel/preset-react": "^7.16.7",
+    "@babel/preset-typescript": "^7.15.0",
+    "@types/react": "^18.0.9",
+    "@types/react-dom": "^18.0.4",
+    "babel-loader": "^8.2.3",
+    "css-loader": "^6.4.0",
+    "file-loader": "^6.2.0",
```
Comment (brauliodiez) [inline 00-boilerplate/package.json:25]: Remove file-loader we will use asset loader

File: 00-boilerplate/package.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    "html-loader": "^3.0.0",
+    "html-webpack-plugin": "^5.4.0",
+    "npm-run-all": "^4.1.5",
+    "rimraf": "^3.0.2",
+    "style-loader": "^3.3.1",
+    "typescript": "^4.4.4",
+    "url-loader": "^4.1.1",
+    "webpack": "^5.59.1",
+    "webpack-cli": "^4.9.1",
+    "webpack-dev-server": "^4.3.1"
```
Comment (brauliodiez) [inline 00-boilerplate/package.json:35]: Update dependencies, install

https://www.npmjs.com/package/npm-check-updates

and execute
ncu -u

then 

npm install

and see if anything brokes O:-)

File: 00-boilerplate/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import { App } from "./app";
+
+ReactDOM.render(
+  <div>
+    <App />
+  </div>,
+  document.getElementById("root")
+);
```
Comment (brauliodiez) [inline 00-boilerplate/src/index.tsx:10]: If you open the console you will get a warning in react 18 you should not use this approach, will send you an updated example

File: 00-boilerplate/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      {
+        test: /\.tsx?$/,
+        exclude: /node_modules/,
+        loader: "babel-loader",
+      },
+      {
+        test: /\.(png|jpg)$/,
+        exclude: /node_modules/,
+        loader: "url-loader",
+      },
```
Comment (brauliodiez) [inline 00-boilerplate/webpack.config.js:29]: you should use the new asset loader from webpack, will include an example

File: 00-boilerplate/readme_es.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# 02 Web Boiler plate
+
+## Resumen
```
Comment (brauliodiez) [inline 00-boilerplate/readme_es.md:3]: Just add the right text to this readme so anybody can understand what is this example for

File: 00-boilerplate/package.json (hunk 3 above)
Comment (carlos-arcos) [reply 00-boilerplate/package.json]: done

File: 00-boilerplate/src/index.tsx (hunk 4 above)
Comment (carlos-arcos) [reply 00-boilerplate/src/index.tsx]: done

File: 00-boilerplate/webpack.config.js (hunk 5 above)
Comment (carlos-arcos) [reply 00-boilerplate/webpack.config.js:28]: done

File: 00-boilerplate/readme_es.md (hunk 6 above)
Comment (carlos-arcos) [reply 00-boilerplate/readme_es.md:3]: done

**Pull request merged**